src/components/label/Label.tsx

```diff
@@ -24,6 +24,8 @@ export interface LabelProps {
     infoIconProps?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
     /** Tooltip text for the info icon */
     infoTooltip?: React.ReactNode;
+    /** Optional props for the label element */
+    labelElProps?: Object;
```

Comment:
We can use a more appropriate and specific type here instead of using a generic `Object`. Either `React.ComponentPropsWithoutRef<'label'>` or `React.LabelHTMLAttributes<HTMLLabelElement>`.

Reply:
Another strategy would be to extend `React.ComponentPropsWithoutRef<'label'>` to allow `<Label>` to take all the normal props of `<label>` instead of having to pass them through an extra prop.

```tsx
export interface LabelProps extends React.ComponentPropsWithoutRef<'label'> {
    //...
}

const Label = ({
    text,
    tooltip,
    infoTooltip,
    infoIconProps,
    showOptionalText,
    hideLabel,
    children,
    ...labelProps
}: LabelProps) => {
    //...

    return (
        <StandardLabel labelContent={labelContent} tooltip={tooltip} {...labelProps}>
            {children}
        </StandardLabel>
    );
};
```